### PR TITLE
SwapShareBackup:Modifying code to delete backup of all users first and then verifying it

### PR DESCRIPTION
**Pull request by snigdha-px**
**What this PR does / why we need it**:
two users have created backup from same namespace and same cluster
user1 created first so its full backup
user2 created second so it incremental backup
now we are deleting user1’s backup first so its in pending state as it waiting for user2's backup to be deleted first
Modification: Deleted backup of both the users first and the verifying the deletion.
**Which issue(s) this PR fixes** (optional)
Closes #https://portworx.atlassian.net/browse/PB-5265

**Special notes for your reviewer**:

s3:https://jenkins.pwx.dev.purestorage.com/job/portworx-backup/job/system-tests/job/byoc/job/px-backup-on-demand-system-test-byoc/3885/

NFS: https://jenkins.pwx.dev.purestorage.com/job/portworx-backup/job/system-tests/job/byoc/job/px-backup-on-demand-system-test-byoc/3886/

### CODE REVIEW

File: tests/backup/backup_share_test.go [hunk 1]
```diff
@@ -4123,11 +4123,15 @@ var _ = Describe("{SwapShareBackup}", func() {
 			ctx, err := backup.GetNonAdminCtx(users[i], commonPassword)
 			log.FailOnError(err, "Fetching non admin ctx")
 			_, err = DeleteBackup(backupName, backupUIDList[i], orgID, ctx)
-			dash.VerifySafely(err, nil, fmt.Sprintf("Verifying backup deletion - %s", backupName))
-			backupDriver := Inst().Backup
-			backupEnumerateReq := &api.BackupEnumerateRequest{
-				OrgId: orgID,
-			}
+			dash.VerifySafely(err, nil, fmt.Sprintf("Deleting backup - %s of user %s", backupName, users[i]))
```
Comment (kphalgun-px) [inline tests/backup/backup_share_test.go:4126]: Is it possible to make use of DeleteBackupAndWait?

File: tests/backup/backup_share_test.go [hunk 2]
```diff
@@ -4123,11 +4123,15 @@ var _ = Describe("{SwapShareBackup}", func() {
...
+		}
+		backupDriver := Inst().Backup
+		backupEnumerateReq := &api.BackupEnumerateRequest{
+			OrgId: orgID,
+		}
+		for i := numberOfUsers - 1; i >= 0; i-- {
+			ctx, err := backup.GetNonAdminCtx(users[i], commonPassword)
```
Comment (kphalgun-px) [inline tests/backup/backup_share_test.go:4133]: Is it possible to make use of DeleteBackupAndWait?

Comment (snigdha-px) [reply tests/backup/backup_share_test.go:4130]: done

Comment (snigdha-px) [reply tests/backup/backup_share_test.go:4130]: https://jenkins.pwx.dev.purestorage.com/job/portworx-backup/job/system-tests/job/byoc/job/px-backup-on-demand-system-test-byoc/3887/

**Review by ak-px — APPROVED** [review]
LGTM

**Review by mkoppal-px — APPROVED** [review]
LGTM

**Review by kshithijiyer-px — APPROVED** [review]
LGTM

**Review by kphalgun-px — APPROVED** [review]
LGTM

**Pull request merged**